src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityFusionReactor.java

```diff
@@ -403,13 +404,24 @@ public boolean isGlobalRenderer() {
         return true;
     }
 
-    static ICustomRenderFast RENDER_HANDLER = new ICustomRenderFast(){
+    static BloomEffectUtil.IBloomRenderFast RENDER_HANDLER = new BloomEffectUtil.IBloomRenderFast(){
+        @Override
+        public int customBloomStyle() {
+            return ConfigHolder.client.shader.fusionBloom.useShader ? ConfigHolder.client.shader.fusionBloom.bloomStyle : -1;
+        }
+
         float lastBrightnessX;
         float lastBrightnessY;
 
         @Override
         @SideOnly(Side.CLIENT)
         public void preDraw(BufferBuilder buffer) {
+            BloomEffect.strength = (float) ConfigHolder.client.shader.fusionBloom.strength;
+            BloomEffect.baseBrightness = (float) ConfigHolder.client.shader.fusionBloom.baseBrightness;
+            BloomEffect.highBrightnessThreshold = (float) ConfigHolder.client.shader.fusionBloom.highBrightnessThreshold;
+            BloomEffect.lowBrightnessThreshold = (float) ConfigHolder.client.shader.fusionBloom.lowBrightnessThreshold;
+            BloomEffect.step = 1;
```

Comment:
Do these need to be reset? Will they "persist" to other bloom calculations or is this localized?

Reply:
This can be interpreted as setting of the current env, which is always reset before rendering of the one pipeline. can be regarded as the `GlStateManager.XXXX`